internal/driver/webhtml.go

```diff
@@ -607,36 +602,58 @@ function viewer(baseUrl, nodes) {
     return str.replace(/([\\\.?+*\[\](){}|^$])/g, '\\$1')
   }
 
-  // Navigate to specified path with current selection reflected
-  // in the named parameter.
-  function navigate(path, param) {
-    // The selection can be in one of two modes: regexp-based or
-    // list-based.  Construct regular expression depending on mode.
-    let re = regexpActive ? search.value : ""
-    if (!regexpActive) {
-      selected.forEach(function(v, key) {
-        if (re != "") re += "|"
-        re += quotemeta(nodes[key])
-      })
-    }
+  // Update id's href to reflect current selection whenever it is
+  // liable to be followed.
+  function makeLinkDynamic(id) {
+    const elem = document.getElementById(id)
+    if (elem == null) return
 
-    const url = new URL(window.location.href)
-    url.pathname = path
-    url.hash = ""
+    // Most links copy current selection into the "f" parameter,
+    // but Refine menu links are different.
+    let param = "f"
+    if (id == "ignore") param = "i"
+    if (id == "hide") param = "h"
+    if (id == "show") param = "s"
+
+    // updater does the actual work.
+    function updater() {
+      const url = new URL(elem.href)
+      url.hash = ""
+
+      // The selection can be in one of two modes: regexp-based or
+      // list-based.  Construct regular expression depending on mode.
+      let re = regexpActive ? search.value : ""
+      if (!regexpActive) {
+        selected.forEach(function(v, key) {
```

Comment:
a more functional/compact way to write this section would be:
```
let re = regexpActive
    ? search.value
    : selected.keys().map(key => quotemeta(nodes[key]).join("|")
```

Reply:
Doesn't work:
"caught TypeError: selected.keys(...).map is not a function"

I think map() does not work Iterator, which is what selected.keys() returns:
https://stackoverflow.com/questions/43885365/using-map-on-an-iterator

I am inclined to stick with forEach; seems clear enough, even if it is not as compact.

Reply:
Oh, right, I meant Array.from(selected.keys()), but mentally filtered that out... It is a suggestion that does not need to deal with two conditions and special treatment of the first item...

Reply:
Yes "Array.from" works. Done.